packages/flutter_tools/lib/src/asset.dart

```diff
@@ -367,18 +385,74 @@ DevFSContent _createAssetManifest(Map<_Asset, List<_Asset>> assetVariants) {
   return new DevFSStringContent(JSON.encode(json));
 }
 
-DevFSContent _createFontManifest(Map<String, dynamic> manifestDescriptor,
-                             bool usesMaterialDesign,
-                             bool includeDefaultFonts) {
+List<Map<String, dynamic>>_parseFonts(
+  Map<String, dynamic> manifestDescriptor,
+  bool usesMaterialDesign,
+  bool includeDefaultFonts,
+  PackageMap packageMap, {
+  String packageName
+}) {
   final List<Map<String, dynamic>> fonts = <Map<String, dynamic>>[];
   if (usesMaterialDesign && includeDefaultFonts) {
     fonts.addAll(_getMaterialFonts(AssetBundle._kFontSetMaterial));
   }
   if (manifestDescriptor != null && manifestDescriptor.containsKey('fonts'))
-    fonts.addAll(manifestDescriptor['fonts']);
-  if (fonts.isEmpty)
-    return null;
-  return new DevFSStringContent(JSON.encode(fonts));
+    if (packageName == null) {
+      fonts.addAll(manifestDescriptor['fonts']);
+    } else {
+      fonts.addAll(_parsePackageFonts(
+        manifestDescriptor['fonts'],
+        packageName,
+        packageMap,
+      ));
+    }
+  return fonts;
+}
+
+/// Prefixes family names and asset paths of fonts included from packages with
+/// 'packages/<package_name>
+///
+/// (e.g., replace {"fonts":[{"asset":"a/bar"}],"family":"bar"} by
+/// {"fonts":[{"asset":"packages/foo/a/bar"}],"family":"packages/foo/bar"})
+List<Map<String, dynamic>> _parsePackageFonts(
```

Comment:
Cf. earlier comment. This method might benefit from an explicit `FontDescriptor` class. Hopefully we wouldn't then have to throw around so many variables of type `Map<String, dynamic>` in one place.